packages/namespace/src/index.ts

```diff
@@ -0,0 +1,277 @@
+/**
+ * Namespace Provider - Factory-based Implementation
+ */
+
+import { createProvider, createBackgroundCommand } from 'computesdk';
+import type { Runtime, ExecutionResult, SandboxInfo, CreateSandboxOptions, FileEntry, RunCommandOptions } from 'computesdk';
+// @ts-expect-error
+import { loadDefaults } from "@namespacelabs/sdk/auth";
+
+/**
+ * Namespace sandbox interface
+ */
+interface NamespaceSandbox {
+  instanceId: string;
+  name: string;
+}
+
+export interface NamespaceConfig {
+  /** Namespace API token - if not provided, will fallback to NSC_TOKEN environment variable */
+  token?: string;
+  /** Virtual CPU cores for the instance */
+  virtualCpu?: number;
+  /** Memory in megabytes for the instance */
+  memoryMegabytes?: number;
+  /** Machine architecture (default: amd64) */
+  machineArch?: string;
+  /** Operating system (default: linux) */
+  os?: string;
+  /** Documented purpose for the instance */
+  documentedPurpose?: string;
+  /** Reason for destroying instances (default: "ComputeSDK cleanup") */
+  destroyReason?: string;
+}
+
+export const getAndValidateCredentials = async (config: NamespaceConfig) => {
+  const tokenSource = await loadDefaults();
+  const issuedToken = await tokenSource.issueToken(5 * 60 * 1000); // 5 minutes
+  const token = config.token || issuedToken;
```

Comment:
@hugosantos we're still accepting a config.token, but falling back to y'all's SDK for the auth side of things here.

in our testing we were seeing soemthing like this: "Error: Session token refresh not yet implemented - requires IAM service integration"

would it be possible to expeirence this on local? or does this mean we've implemented wrong?